coalib/parsing/LineParser.py

```diff
@@ -148,12 +157,19 @@ def __get_section_name(self, line):
 
         return ''
 
-    def __extract_keys_and_value(self, line):
+    def __extract_keys_and_value(self, line, delimiters):
+        """
+        This method extracts the keys and values from the give string by
+        splitting them based on the delimiters provided.
+        :param line:       The input string.
+        :param delimiters: A list of delimiters to split the strings on.
+        """
         key_part, value = self.__separate_by_first_occurrence(
             line,
-            self.key_value_delimiters,
+            delimiters,
             True,
-            True)
+            # To return empty value in case of no append delimiter is found.
+            delimiters == self.key_value_define_delimiters)
```

Comment:
this is rather dirty I think and hurts readability, just make another bool parameter for this in the signature and explain it properly.